src/rules/attribute-names.ts

```diff
@@ -49,14 +67,37 @@ const rule: Rule.RuleModule = {
                 });
               }
             } else {
-              if (
-                propConfig.attributeName.toLowerCase() !==
-                propConfig.attributeName
-              ) {
-                context.report({
-                  node: propConfig.expr ?? propConfig.key,
-                  messageId: 'casedAttribute'
-                });
+              if (style === 'none') {
+                if (propConfig.attributeName !== prop.toLowerCase()) {
```

Comment:
you can reduce the duplication here and go back to having a single message by doing something like this:

```ts
    messages: {
      casedAttribute:
        'Attributes are case-insensitive. This attribute should be written as a {{style}} name',
      casedPropertyWithoutAttribute:
        'Property has non-lowercase casing but no attribute. It should ' +
        'instead have an explicit `attribute` set to the {{style}} ' +
        'name'
   }
```

```ts
let styleName: string;
let styledKey: string;

switch (style) {
  case 'snake':
    styleName = 'snake_case';
    styledKey = toSnakeCase(prop);
    break;
  case 'dash':
    styleName = 'dash-case';
    styledKey = toDashCase(prop);
    break;
  default:
    styleName = 'lower case';
    styledKey = prop.toLowerCase();
    break;
}

if (propConfig.attributeName !== styledKey) {
  context.report({
    node: propConfig.expr ?? propConfig.key,
    messageId: 'casedAttribute',
    data: {style: styleName}
  });
}
```

similarly, you would then change the other message to be more like this, too:

```ts
              if (prop.toLowerCase() !== prop) {
                context.report({
                  node: propConfig.key,
                  messageId: 'casedPropertyWithoutAttribute',
                  data: {style: styleName}
                });
```

Reply:
I pushed an adapted version of that, taking in account that no style defined is not the same rule as `{"style": "none"}`
See https://github.com/43081j/eslint-plugin-lit/pull/207#discussion_r1696864454